test/e2e/user_management_test.go

```diff
@@ -203,6 +212,39 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 			metricsAssertion.WaitForMetricDelta(UserAccountsMetric, 0, "cluster_name", s.member2Await.ClusterName) // no user on member2
 		})
 	})
+
+	s.T().Run("test deactivating state set OK", func(t *testing.T) {
+		// Reset configuration back to 3 days
+		s.hostAwait.UpdateHostOperatorConfig(
+			test.AutomaticApproval().Enabled(),
+			test.Deactivation().DeactivatingNotificationDays(3))
+
+		config := s.hostAwait.GetHostOperatorConfig()
+		require.Equal(s.T(), 3, config.Spec.Deactivation.DeactivatingNotificationDays)
+
+		userSignupMember1, murMember1 := s.createAndCheckUserSignup(true, "usertostartdeactivating",
+			"usertostartdeactivating@redhat.com", s.memberAwait, ApprovedByAdmin()...)
+
+		// Get the provisioned account's tier
+		baseTier, err := s.hostAwait.WaitForNSTemplateTier("base")
+		require.NoError(t, err)
+
+		// We cannot wait days for testing deactivation so for the purposes of the e2e tests we use a hack to change the
+		// provisioned time to a time far enough in the past to trigger the deactivation process. Subtracting the given
+		// period from the current time and setting this as the provisioned time should test the behaviour of the
+		// deactivation controller reconciliation.
+		tierDeactivationDuration := time.Duration(baseTier.Spec.DeactivationTimeoutDays + 1) * time.Hour * 24
+		murMember1, err = s.hostAwait.UpdateMasterUserRecordStatus(murMember1.Name, func(mur *v1alpha1.MasterUserRecord) {
+			mur.Status.ProvisionedTime = &metav1.Time{Time: time.Now().Add(-tierDeactivationDuration)}
+		})
+		require.NoError(s.T(), err)
+		s.T().Logf("masteruserrecord '%s' provisioned time adjusted to %s", murMember1.Name,
+			murMember1.Status.ProvisionedTime.String())
+
+		// The user should be set to deactivating, but not deactivated
+		_, err = s.hostAwait.WaitForUserSignup(userSignupMember1.Name, wait.UntilUserSignupHasConditions(Deactivating()...))
```

Comment:
After the wait let's verify the usersignup resources: `VerifyResourcesProvisionedForSignup(...)` to make sure the user is still active.
Also, shouldn't we check that the Deactivating Notificaton resource is created? It will be in a failing state but we at least could verify it's created, right?

Reply:
So this is not so easy as it seems.  In the Notification controller, if running within the e2e tests environment the notification is always treated as sent successfully:

```
	// if the environment is set to e2e do not attempt sending via mailgun
	if r.config.GetEnvironment() != "e2e-tests" {
		// Send the notification via the configured delivery service
		err = r.deliveryService.Send(notCtx, notification)
		if err != nil {
			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, notification,
				r.setStatusNotificationDeliveryError, err, "failed to send notification")
		}
		reqLogger.Info("Notification has been sent")
	} else {
		reqLogger.Info("Notification has been skipped")
	}

	return reconcile.Result{
		Requeue:      true,
		RequeueAfter: r.config.GetDurationBeforeNotificationDeletion(),
	}, r.updateStatus(reqLogger, notification, r.setStatusNotificationSent)
```


What this means, is that the Notification resource gets deleted fasted than we have the check that it was created:

```
{"level":"info","ts":1619741085.5813034,"logger":"controller_notification","msg":"Reconciling Notification","Request.Namespace":"sbryzak-host-30090225","Request.Name":"usertostartdeactivating-provisioned-sh9tx"}
{"level":"info","ts":1619741085.5813544,"logger":"controller_notification","msg":"the Notification is sent so we can deal with its deletion","Request.Namespace":"sbryzak-host-30090225","Request.Name":"usertostartdeactivating-provisioned-sh9tx"}
{"level":"info","ts":1619741085.581392,"logger":"controller_notification","msg":"the Notification has been sent for a longer time than the 'durationBeforeNotificationDeletion', so it's ready to be deleted","Request.Namespace":"sbryzak-host-30090225","Request.Name":"usertostartdeactivating-provisioned-sh9tx","durationBeforeNotificationDeletion":"5s"}
{"level":"info","ts":1619741085.594358,"logger":"controller_notification","msg":"Reconciling Notification","Request.Namespace":"sbryzak-host-30090225","Request.Name":"usertostartdeactivating-provisioned-sh9tx"}
```

I don't have an easy solution for checking whether the Notification resource was ever created, any ideas for how we might do this?

Reply:
Ok. Then let's skip the notification check part.

Reply:
Actually, one idea did occur to me in the meantime.  What do you think of handling it this way:

https://github.com/codeready-toolchain/host-operator/pull/393/files#diff-7a61df0b58a303589124157586cb34986301fcbb8e75a4d3fe0f5efb1543bc15R127

Reply:
I'm fine with this if it works :)

Reply:
Ah geez, turns out there's other tests that actually wait for the notification to be deleted.  Seems like there's no easy fix after all, I'll revert this change and remove the notification check.